pkg/kv/kvserver/replica_learner_test.go

```diff
@@ -1533,3 +1533,276 @@ func TestMergeQueueSeesLearnerOrJointConfig(t *testing.T) {
+	receiverRebalancingSentBytesBefore := getFirstStoreMetric(t, tc.Server(1), "range.snapshots.rebalancing.sent-bytes")
+	receiverRebalancingRcvdBytesBefore := getFirstStoreMetric(t, tc.Server(1), "range.snapshots.rebalancing.rcvd-bytes")
+
+	senderRecoverySentBytesBefore := getFirstStoreMetric(t, tc.Server(0), "range.snapshots.recovery.sent-bytes")
+	senderRecoveryRcvdBytesBefore := getFirstStoreMetric(t, tc.Server(0), "range.snapshots.recovery.rcvd-bytes")
+	receiverRecoverySentBytesBefore := getFirstStoreMetric(t, tc.Server(1), "range.snapshots.recovery.sent-bytes")
+	receiverRecoveryRcvdBytesBefore := getFirstStoreMetric(t, tc.Server(1), "range.snapshots.recovery.rcvd-bytes")
+
+	senderUnknownSentBytesBefore := getFirstStoreMetric(t, tc.Server(0), "range.snapshots.unknown.sent-bytes")
+	senderUnknownRcvdBytesBefore := getFirstStoreMetric(t, tc.Server(0), "range.snapshots.unknown.rcvd-bytes")
+	receiverUnknownSentBytesBefore := getFirstStoreMetric(t, tc.Server(1), "range.snapshots.unknown.sent-bytes")
+	receiverUnknownRcvdBytesBefore := getFirstStoreMetric(t, tc.Server(1), "range.snapshots.unknown.rcvd-bytes")
+
+	senderTotalSentBytesBefore := getFirstStoreMetric(t, tc.Server(0), "range.snapshots.sent-bytes")
+	senderTotalRcvdBytesBefore := getFirstStoreMetric(t, tc.Server(0), "range.snapshots.rcvd-bytes")
+	receiverTotalSentBytesBefore := getFirstStoreMetric(t, tc.Server(1), "range.snapshots.sent-bytes")
+	receiverTotalRcvdBytesBefore := getFirstStoreMetric(t, tc.Server(1), "range.snapshots.rcvd-bytes")
+
+	g := ctxgroup.WithContext(ctx)
+	g.GoCtx(func(ctx context.Context) error {
+		_, err := tc.AddVoters(scratchStartKey, tc.Target(1))
+		return err
+	})
+
+	// Wait until the snapshot starts, which happens after the learner has been
+	// added.
+	<-blockUntilSnapshotCh
+
+	// Unblock the snapshot and let the learner finish receiving the snapshot
+	close(blockSnapshotsCh)
+	require.NoError(t, g.Wait())
+
+	// Asserts about the snapshot sender replica
+	senderRebalancingSentBytesAfter := getFirstStoreMetric(t, tc.Server(0), "range.snapshots.rebalancing.sent-bytes")
+	senderRebalancingRcvdBytesAfter := getFirstStoreMetric(t, tc.Server(0), "range.snapshots.rebalancing.rcvd-bytes")
+
+	senderRecoverySentBytesAfter := getFirstStoreMetric(t, tc.Server(0), "range.snapshots.recovery.sent-bytes")
+	senderRecoveryRcvdBytesAfter := getFirstStoreMetric(t, tc.Server(0), "range.snapshots.recovery.rcvd-bytes")
+
+	senderUnknownSentBytesAfter := getFirstStoreMetric(t, tc.Server(0), "range.snapshots.unknown.sent-bytes")
+	senderUnknownRcvdBytesAfter := getFirstStoreMetric(t, tc.Server(0), "range.snapshots.unknown.rcvd-bytes")
+
+	senderTotalSentBytesAfter := getFirstStoreMetric(t, tc.Server(0), "range.snapshots.sent-bytes")
+	senderTotalRcvdBytesAfter := getFirstStoreMetric(t, tc.Server(0), "range.snapshots.rcvd-bytes")
+
+	require.True(t, senderRebalancingSentBytesAfter > senderRebalancingSentBytesBefore)
```

Comment:
nit: For these you can do require.Equal, require.Greater rather than require.True.

Reply:
Updated!